### DISCUSSION TITLE
How do I remove mobx out of bounds warning?

### DISCUSSION THREAD

**Discussion by ivan-palatov** [mobxjs/mobx#3429] [Q&A] [answered]
I know what you all are probably thinking, "just fix your code, you are doing something wrong with it". But it is not the case, I work with geo spacial data and use deck.gl to render stuff on map, a lot of stuff.
I store the data I work it in a mobx store and than pass it to deck.gl when I need to. The problem is, deck.gl reads all of the coordinates, which are arrays, assuming that there might be a third one [lat, lng and alt?], but my coordinates are arrays of 2 [lat, lng].

And every time deck.gl tries to read it, a warning is thrown in the console. And it would've been fine, if not for blocking the thread for any number of seconds at a time. Did I already tell you that there is A LOT of data in the app I work on?

Sure, I could use slice on my arrays when I pass them to deck.gl and I could probably just make a computed that would add a third coordinate of 0 to all of position arrays, but the latter would increase the memory usage, although slightly and the former really fills the code with unnecessary bloat.

I'd much rather turn the warnings off, is it possible to do? Is there a better way the the ones described above to do so?

**Comment by kubk** [comment, marked as answer]
I don't have an experience with deck.gl, but I assume you can call `toJS(coordinates)` before passing them to the library unless the library modifies them somehow and you need to react on it

**Reply by ivan-palatov** [reply]
Thanks, I think that would work. Didn't notice that was a thing in the docs

**Reply by urugator** [reply]
Note `toJS` is basically just slightly less efficient `slice()` in this case.

**Reply by mweststrate** [reply]
Make sure to check perf though, even the data needs to be converted often and there is loads of data, it might hit pretty badly on the GC without significant upside beyond suppressing the error.

**Comment by mweststrate** [comment]
Patch-package on either deck.gl to prevent the warning, or mobx to supress
it?

**Comment by urugator** [comment]
Perhaps stupid question, but do these arrays have to be observable? Do you ever modify the array itself?

**Reply by ivan-palatov** [reply]
Sadly, yes, they are modifiable and are used in other places in the app.
